language/move-symbol-pool/src/pool.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#[forbid(unsafe_code)]
+
+type SymbolData = Box<str>;
+
+/// An ordered, unique set of strings.
+///
+/// To save on heap space, insertion into the set iterates over its elements to
+/// enforce uniqueness, and so is `O(n)`.
+pub(crate) struct Pool(Vec<SymbolData>);
+
+impl Pool {
+    pub(crate) fn new() -> Self {
+        Self { 0: vec![] }
+    }
+
+    pub(crate) fn get(&self, idx: usize) -> *const SymbolData {
+        &self.0[idx]
+    }
+
+    pub(crate) fn insert(&mut self, s: &str) -> usize {
```

Comment:
This seems... bad
Why not just use string_cache's table layout? I know you want to stay simple for the version version... but this is a bit too simple for my tastes

Reply:
Yeah, good call, I hadn't considered taking another look after making this latest change. Will do!

Reply:
OK, the latest iteration is missing some tests that I'll add later today, but it uses string_cache's approach to storage 👍